cpp/dolfinx/fem/Form.h

```diff
@@ -277,9 +277,9 @@ class Form
   /// Get the list of (cell_index, local_facet_index) pairs for the ith
   /// integral (kernel) for the exterior facet domain type
   /// @param[in] i Integral ID, i.e. (sub)domain index
-  /// @return List of (cell_index, local_facet_index) pairs
-  const std::vector<std::pair<std::int32_t, int>>&
-  exterior_facet_domains(int i) const
+  /// @return List of (cell_index, local_facet_index) pairs. Storage is row
```

Comment:
The shape needs to be given for the layout of the returned data to be fully defined.